liquigraph-core/src/main/java/org/liquigraph/core/io/GraphJdbcConnector.java

```diff
@@ -25,13 +26,21 @@
 
 public class GraphJdbcConnector implements LiquigraphJdbcConnector {
 
+    /**
+     * Acquires a new connection to the configured instance
+     * and tries to lock it (fail-fast).
+     *
+     * @see {@link LockableConnection}
+     * @param configuration Liquigraph settings
+     * @return JDBC connection
+     */
     @Override
     public final Connection connect(Configuration configuration) {
         try {
             Class.forName("org.neo4j.jdbc.Driver");
             Connection connection = DriverManager.getConnection(makeUri(configuration));
             connection.setAutoCommit(false);
-            return connection;
+            return new LockableConnection(connection);
```

Comment:
If the lock cannot be acquired and an exception is thrown, the original connection is not closed. Instead of the simple `return` statement, something like:
```java
try {
     return new LockableConnection(connection);
} catch (RuntimeException e) {
    connection.close();
    throw e;
}
```

Or manage it in the constructor of `LockableConnection`, maybe?